core/src/main/scala/caliban/execution/Field.scala

```diff
@@ -97,20 +97,21 @@ object Field {
     variableDefinitions: List[VariableDefinition],
     variableValues: Map[String, InputValue]
   ): Map[String, InputValue] = {
-    def resolveVariable(value: InputValue): InputValue =
+    def resolveVariable(value: InputValue): Option[InputValue] =
       value match {
-        case InputValue.ListValue(values)   => InputValue.ListValue(values.map(resolveVariable))
+        case InputValue.ListValue(values)   =>
+          Some(InputValue.ListValue(values.flatMap(resolveVariable)))
         case InputValue.ObjectValue(fields) =>
-          InputValue.ObjectValue(fields.map { case (k, v) => k -> resolveVariable(v) })
+          Some(InputValue.ObjectValue(fields.flatMap { case (k, v) => resolveVariable(v).map(k -> _) }))
         case InputValue.VariableValue(name) =>
-          (for {
-            definition  <- variableDefinitions.find(_.name == name)
-            defaultValue = definition.defaultValue getOrElse NullValue
-            value        = variableValues.getOrElse(name, defaultValue)
-          } yield value) getOrElse NullValue
-        case value: Value                   => value
+          for {
+            definition <- variableDefinitions.find(_.name == name)
+            value      <- variableValues.get(name).orElse(definition.defaultValue)
```

Comment:
nit 💅 : getOrElse

Reply:
`definition.defaultValue` is an `Option` so `getOrElse` is not applicable?

Reply:
Yeah I think your right

Reply:
But so is `variableValues.get(name)` since you're getting it from a map, right?

Reply:
@frekw `getOrElse` expects an `A`, but `defaultValue` is an `Option[A]`. It's `orElse`, not `getOrElse`.

Reply:
I meant `getOrElse` from Map :) https://scastie.scala-lang.org/jywLcoZARNG8KOrI7LDHqw

Reply:
But in either case, doesn't matter much.

Reply:
No, `b` is of type `Any` if you do that :P

https://scastie.scala-lang.org/YjZ63y0XRXSxAMGDC6K65Q

Reply:
😬

Reply:
And also 🤦  I guess